hosters/hetzner-dedicated/hetzner-dedicated-wipe-and-install-nixos.sh

```diff
@@ -237,51 +336,76 @@ cat > /mnt/etc/nixos/configuration.nix <<EOF
   # We do not worry about plugging disks into the wrong machine because
   # we will never exchange disks between machines.
   environment.etc."mdadm.conf".text = ''
-    HOMEHOST hetzner
+    HOMEHOST $HOSTNAME
   '';
+
   # The RAIDs are assembled in stage1, so we need to make the config
   # available there.
   boot.initrd.mdadmConf = config.environment.etc."mdadm.conf".text;
 
   # Network (Hetzner uses static IP assignments, and we don't use DHCP here)
   networking.useDHCP = false;
-  networking.interfaces."$NIXOS_INTERFACE".ipv4.addresses = [
-    {
-      address = "$IP_V4";
-      prefixLength = 24;
-    }
-  ];
-  networking.interfaces."$NIXOS_INTERFACE".ipv6.addresses = [
-    {
-      address = "$IP_V6";
-      prefixLength = 64;
-    }
-  ];
-  networking.defaultGateway = "$DEFAULT_GATEWAY";
-  networking.defaultGateway6 = { address = "fe80::1"; interface = "$NIXOS_INTERFACE"; };
-  networking.nameservers = [ "8.8.8.8" ];
+
+  networking.interfaces."$NET_IFACE" = {
+    ipv4 = {
+      addresses = [{
+        # Server main IPv4 address
+        address = "$IP_V4";
+        prefixLength = 24;
+      }];
+
+      routes = [
+        # Default IPv4 gateway route
+        {
+          address = "0.0.0.0";
+          prefixLength = 0;
+          via = "$DEFAULT_GATEWAY";
+        }
+      ];
+    };
+
+    ipv6 = {
+      addresses = [{
+        address = "$IP_V6";
+        prefixLength = 64;
+      }];
+
+      # Default IPv6 route
+      routes = [{
+        address = "::";
+        prefixLength = 0;
+        via = "fe80::1";
+      }];
+    };
+  }
```

Comment:
Missing semicolon.

Reply:
@srid sorry to ping you here, but did you have any dns issues when installing NixOS using this script? For me somehow the dns does not work, other than that the scrips runs fine, except the comments I've put above.

Reply:
Not the last time I tried it (in February), but maybe things have changed on Hetzner recently.